components/pedigree/resources/src/main/resources/pedigree/externalEndpoints.js

```diff
@@ -105,6 +105,18 @@ define([
             return XWiki.contextPath + "/rest/patients/" + patientId + "?method=DELETE";
         },
 
+        getPatientUpdateURL: function(patientId) {
+            // note: have to use "?method=PUT" because of pecularities of how prototype Ajax.Request works
+            //       see: http://api.prototypejs.org/ajax/
+            //       see: https://stackoverflow.com/questions/4476810/is-it-really-possible-to-call-http-put-using-prototype
+            return XWiki.contextPath + "/rest/patients/" + patientId + "?method=PUT";
```

Comment:
we can add `this.patientService = XWiki.contextPath + "/rest/patients";` to have one point where we store REST URL and use it here like `return this.patientService + patientId + "?method=PUT"`